### PR TITLE
Added main menu

### PR DESCRIPTION
**Pull request by noah-nm**
Currently I have controller polling done in game. This makes it easyier (imo) to exit games plus it's cleaner cuz all the logic is done in game.

### CODE REVIEW

File: src/App.java
Comment (noah-nm) [inline src/App.java]: I was going to use this for something but decided agianst it. Will remove before merge.

File: src/menus/MainMenu.java
Comment (HoodieRocks) [inline src/menus/MainMenu.java]: This return line could probably be removed, but it doesn't affect the code at all so it can be merged in.

Comment (noah-nm) [reply src/menus/MainMenu.java]: ya idk why I did that, I do not know javadocs lol.

File: src/menus/Menu.java
Comment (HoodieRocks) [inline src/menus/Menu.java:14]: I feel like it's not necessary to pass in the controllers for a total of 3 lines of code. Controller polling should probably be done always if we want to use it as the main control scheme

Comment (noah-nm) [reply src/menus/Menu.java:14]: but polling has done be done through controller and input checking has to be done through gamepads no?

Comment (noah-nm) [reply src/menus/Menu.java:14]: what are you suggesting?

Comment (HoodieRocks) [reply src/menus/Menu.java:14]: What I mean is controller polling can be done in the main while loop instead of per-menu, I just want to minimize the amount of boilerplate required for each game, even if it is only 3 lines

Comment (noah-nm) [reply src/menus/Menu.java:14]: okay lol i see my issue now. I was doing this before and it didn't seem to work because it wouldn't exit because it was just one game and polling in the loop in App. I think we will need another screen to test it properly. Forgive me im tired.

Comment (HoodieRocks) [reply src/menus/Menu.java:14]: All good, the functionality is correct and this is just a nitpick so I'll just approve it. I'll let you merge it and if you want to add that functionality in before you merge just re-request a review

**Review by HoodieRocks — APPROVED** [review]
LGTM

File: src/menus/Menu.java
Comment (noah-nm) [reply src/menus/Menu.java:14]: After some playing around here is the issue. If we have a while loop in the game that doesn't poll controllers, the game can never actually take input because the for loop in the App() while loop that polls controllers doesn't actually get input since the game while loop is running and not polling. So the way I see it we eaither think of a different way to exit games or keep it the way it is. LMK if im missing somthing.

Comment (HoodieRocks) [reply src/menus/Menu.java:14]: I think the way we exit games right now is probably fine. It'd be a lot of effort for something so minuscule so I think its best to just let there be boilerplate rather than devise a whole new plan for exiting

LGTM

**Pull request merged**
